frontend/lib/components/TextEditor/TextEditor.tsx

```diff
@@ -52,24 +70,79 @@ export const TextEditor = (): JSX.Element => {
         }),
         BrainMention,
         FocusSearchBar,
+        AIHighlight.configure({
+          HTMLAttributes: {
+            class: styles.ai_highlight,
+          },
+        }),
       ],
       content: defaultContent,
       immediatelyRender: false,
       autofocus: true,
     },
     [items.length]
   );
+
+  const toggleSearchBar = () => {
+    if (searchBarOpen) {
+      setSearchBarOpen(false);
+      editor?.commands.focus();
+    } else {
+      setSearchBarOpen(true);
+      searchEditorRef.current?.commands.focus();
+    }
+  };
+
   if (!editor) {
     return <></>;
   }
 
   return (
     <div className={styles.main_container}>
       <div className={styles.editor_wrapper}>
-        <Toolbar searchBarEditor={searchEditorRef.current} editor={editor} />
+        <Toolbar toggleSearchBar={toggleSearchBar} editor={editor} />
+        <div>
+          <BubbleMenu
+            shouldShow={() => {
+              return editor.isActive("aiHighlight", { type: "ai" });
```

Comment:
can you store aiHighlight (variable name) in a variable, as it's used several times :)